cmd/htcondor-api/token_fetch.go

```diff
@@ -0,0 +1,542 @@
+					IssuerURL:    config.IssuerURL,
+				}, nil
+			}
+
+			var errorResp struct {
+				Error            string `json:"error"`
+				ErrorDescription string `json:"error_description"`
+			}
+
+			if err := json.Unmarshal(body, &errorResp); err == nil {
+				if errorResp.Error == "authorization_pending" {
+					fmt.Print(".")
+					continue
+				}
+				return nil, fmt.Errorf("token error: %s - %s", errorResp.Error, errorResp.ErrorDescription)
+			}
+
+			return nil, fmt.Errorf("unexpected response: %s", string(body))
+		}
+	}
+}
+
+// refreshToken refreshes an access token using a refresh token
+func refreshToken(tokenInfo *TokenInfo, _ *TokenFetchConfig) (*TokenInfo, error) {
+	tokenURL := tokenInfo.IssuerURL + "/mcp/oauth2/token"
+
+	data := url.Values{}
+	data.Set("grant_type", "refresh_token")
+	data.Set("refresh_token", tokenInfo.RefreshToken)
+	data.Set("client_id", tokenInfo.ClientID)
+	data.Set("client_secret", tokenInfo.ClientSecret)
+
+	ctx := context.Background()
+	req, err := http.NewRequestWithContext(ctx, "POST", tokenURL, strings.NewReader(data.Encode()))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	client := &http.Client{Timeout: 30 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			fmt.Fprintf(os.Stderr, "Warning: failed to close response body: %v\n", err)
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("refresh failed: %s", string(body))
+	}
+
+	var tokenResult struct {
+		AccessToken  string `json:"access_token"`
+		RefreshToken string `json:"refresh_token"`
+		TokenType    string `json:"token_type"`
+		ExpiresIn    int    `json:"expires_in"`
+		Scope        string `json:"scope"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&tokenResult); err != nil {
+		return nil, err
+	}
+
+	scopes := strings.Split(tokenResult.Scope, " ")
+	expiresAt := time.Now().Add(time.Duration(tokenResult.ExpiresIn) * time.Second)
+
+	// Use new refresh token if provided, otherwise keep the old one
+	refreshToken := tokenResult.RefreshToken
+	if refreshToken == "" {
+		refreshToken = tokenInfo.RefreshToken
+	}
+
+	return &TokenInfo{
+		TrustDomain:  tokenInfo.TrustDomain,
+		ClientID:     tokenInfo.ClientID,
+		ClientSecret: tokenInfo.ClientSecret,
+		AccessToken:  tokenResult.AccessToken,
+		RefreshToken: refreshToken,
+		TokenType:    tokenResult.TokenType,
+		ExpiresAt:    expiresAt,
+		Scopes:       scopes,
+		IssuerURL:    tokenInfo.IssuerURL,
+	}, nil
+}
+
+// getTokenInfoPath returns the path to the token info file
+func getTokenInfoPath() string {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		homeDir = "/tmp"
+	}
+	condorDir := filepath.Join(homeDir, ".condor")
+	return filepath.Join(condorDir, "token_info")
+}
+
+// getAccessTokenPath returns the path where access token should be saved
```

Comment:
We need to scan the entire directory to check for a specific token, not just the default name.